Source/ROLib/Modules/ROLAnimateEngineHeat.cs

```diff
@@ -0,0 +1,216 @@
+        [KSPField]
+        public float maxHeat = 2400;
+
+        //maxStoredHeat
+        //storedHeat will not go beyond this, sets retention period for maximum glow
+        [KSPField]
+        public float maxStoredHeat = 3600;
+
+        //curve to adjust heat dissipation; should generally expel heat faster when hotter
+        [KSPField]
+        public FloatCurve heatDissipationCurve = new FloatCurve();
+
+        //the heat-output curve for an engine (varies with thrust/throttle), in case it is not linear
+        [KSPField]
+        public FloatCurve heatAccumulationCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve redCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve blueCurve = new FloatCurve();
+
+        [KSPField]
+        public FloatCurve greenCurve = new FloatCurve();
+
+        [KSPField]
+        public string engineID = "Engine";
+
+        [KSPField]
+        public String meshName = String.Empty;
+
+        [KSPField(isPersistant = true)]
+        public float currentHeat = 0;
+
+        [KSPField]
+        public bool useThrottle;
+
+        int shaderEmissiveID;
+
+        private ModuleEngines engineModule;
+
+        private Renderer[] animatedRenderers;
+
+        private Color emissiveColor = new Color(0f, 0f, 0f, 1f);
+
+        public override void OnAwake()
+        {
+            base.OnAwake();
+            heatDissipationCurve.Add(0f, 0.2f);
+            heatDissipationCurve.Add(1f, 1f);
+
+            heatAccumulationCurve.Add(0f, 0f);
+            heatAccumulationCurve.Add(1f, 1f);
+            
+            redCurve.Add(0f, 0f);
+            redCurve.Add(1f, 1f);
+
+            blueCurve.Add(0f, 0f);
+            blueCurve.Add(1f, 1f);
+
+            greenCurve.Add(0f, 0f);
+            greenCurve.Add(1f, 1f);
+
+            shaderEmissiveID = Shader.PropertyToID("_EmissiveColor");
+        }
+
+        public override void OnStart(StartState state)
+        {
+            base.OnStart(state);
+            initialize();
+        }
+
+        public void FixedUpdate()
+        {
+            if (!HighLogic.LoadedSceneIsFlight) { return; }
+            updateHeat();
+        }
+
+        private void initialize()
+        {
+            locateAnimatedTransforms();
+            locateEngineModule();
+        }
+
+        public void reInitialize()
+        {
+            animatedRenderers = null;
+            engineModule = null;
+            initialize();
+        }
+
+        private void locateEngineModule()
+        {
+            engineModule = null;
+            ModuleEngines[] engines = part.GetComponents<ModuleEngines>();
+            int len = engines.Length;
+            for (int i = 0; i < len; i++)
+            {
+                if (engines[i].engineID == engineID)
+                {
+                    engineModule = engines[i];
+                }
+            }
```

Comment:
I don't mind LINQ.  Consider  `engineModule = part.GetComponents<ModuleEngines>().Where(x => x.engineID == engineID).FirstOrDefault()` instead for the entire block.